src/include/network/network_io_wrapper.h

```diff
@@ -37,6 +39,25 @@ class NetworkIoWrapper {
     RestartState();
   }
 
+  explicit NetworkIoWrapper(const std::string &ip_address, uint16_t port)
+      : sock_fd_(socket(AF_INET, SOCK_STREAM, 0)),
```

Comment:
This does not check the error code of socket.

Reply:
In general, this function seems to be a hack. Is there a reason to not create the socket from whatever is calling this?